### PR TITLE
Validate Burst Pairs

### PR DESCRIPTION
**Pull request by AndrewPlayer3**
This PR adds validation for burst pairs. It ensures that the reference and secondary bursts share a common burst id and polarization. It also checks if the polarization is either VV or HH, as those are all we support at the moment. ~~, and that they occurred after ASF began storing bursts. It also ensures that the bursts are not over areas that do not have DEM coverage, including Azerbaijan, Armenia, and large water bodies.~~

### CODE REVIEW

File: images/coverage.svg
Comment (jtherrmann) [inline images/coverage.svg:1]: I assume this was accidentally committed?

File: src/hyp3_isce2/burst.py [hunk 1]
```diff
@@ -380,3 +380,40 @@ def get_burst_params(scene_name: str) -> BurstParams:
...
+        raise ValueError(
+            f'The reference and secondary polarizations are not the same: {ref_polarization} and {sec_polarization}.'
+        )
+
+    if ref_polarization != "VV" and ref_polarization != "HH":
+        raise ValueError(
+            f'{ref_polarization} polarization is not currently supported, only VV and HH.'
+        )
+
+    return None
```
Comment (jtherrmann) [inline src/hyp3_isce2/burst.py:419]: Not needed, this is the default behavior.
```suggestion

    return None
```

File: src/hyp3_isce2/burst.py [hunk 2]
```diff
@@ -380,3 +380,40 @@ def get_burst_params(scene_name: str) -> BurstParams:
         polarization=results[0].properties['polarization'],
         burst_number=results[0].properties['burst']['burstIndex'],
     )
+
+
+def validate_bursts(reference_name, secondary_name):
```
Comment (jtherrmann) [inline src/hyp3_isce2/burst.py:385]:
```suggestion
def validate_bursts(reference_name: str, secondary_name: str) -> None:
```

File: src/hyp3_isce2/burst.py [hunk 3]
```diff
@@ -380,3 +380,40 @@ def get_burst_params(scene_name: str) -> BurstParams:
...
+    sec_split = secondary_name.split('_')
+
+    ref_burst_id = ref_split[1]
+    sec_burst_id = sec_split[1]
+    ref_polarization = ref_split[4]
+    sec_polarization = sec_split[4]
+
+    if ref_burst_id != sec_burst_id:
+        raise ValueError(
+            f'The reference and secondary bursts do not share a common BurstID: {ref_burst_id} and {sec_burst_id}.'
```
Comment (jtherrmann) [inline src/hyp3_isce2/burst.py:406]: I think this wording makes it more clear that the two values are the reference and secondary burst IDs.
```suggestion
            f'The reference and secondary burst IDs are not the same: {ref_burst_id} and {sec_burst_id}.'
```

File: src/hyp3_isce2/burst.py [hunk 4]
```diff
@@ -380,3 +380,40 @@ def get_burst_params(scene_name: str) -> BurstParams:
...
+
+    Returns:
+        None
+    """
+
+    ref_split = reference_name.split('_')
+    sec_split = secondary_name.split('_')
+
+    ref_burst_id = ref_split[1]
+    sec_burst_id = sec_split[1]
```
Comment (jtherrmann) [inline src/hyp3_isce2/burst.py:400]:
```suggestion
    sec_burst_id = sec_split[1]

```

File: src/hyp3_isce2/burst.py [hunk 5]
```diff
@@ -380,3 +380,40 @@ def get_burst_params(scene_name: str) -> BurstParams:
...
+    """Check whether the reference and secondary bursts are valid.
+
+    Args:
+        reference_scene: The reference burst name.
+        secondary_scene: The secondary burst name.
+
+    Returns:
+        None
+    """
+
```
Comment (jtherrmann) [inline src/hyp3_isce2/burst.py:395]:
```suggestion
```

File: src/hyp3_isce2/dem.py [hunk 6]
```diff
@@ -94,6 +94,7 @@ def download_dem_for_isce2(
     Returns:
         The path to the downloaded DEM.
     """
+
```
Comment (jtherrmann) [inline src/hyp3_isce2/dem.py:97]: I don't think this blank line is needed, unless it's to satisfy a linter.
```suggestion
```

File: tests/test_burst.py [hunk 7]
```diff
@@ -168,3 +168,28 @@ def test_get_burst_params_multiple_results():
...
+
+
+def test_validate_bursts():
+    try:
+        burst.validate_bursts(  # valid burst pair
+            'S1_030349_IW1_20230808T171601_VV_4A37-BURST',
+            'S1_030349_IW1_20230820T171602_VV_5AC3-BURST'
+        )
+    except ValueError as e:
+        assert False, f'Unexpected exception for valid burst pair: {e}'
```
Comment (jtherrmann) [inline tests/test_burst.py:180]: No need to try/except here, the intention is clear when reading the code and the test will fail if an exception is raised by the function.
```suggestion
    burst.validate_bursts(
        'S1_030349_IW1_20230808T171601_VV_4A37-BURST',
        'S1_030349_IW1_20230820T171602_VV_5AC3-BURST'
    )
```

File: tests/test_burst.py [hunk 8]
```diff
@@ -168,3 +168,28 @@ def test_get_burst_params_multiple_results():
...
+    with pytest.raises(ValueError, match=r'.*polarizations are not the same.*'):  # different polarizations
```
Comment (jtherrmann) [inline tests/test_burst.py:181]: The code is clear without the comment.
```suggestion
    with pytest.raises(ValueError, match=r'.*polarizations are not the same.*'):
```

File: tests/test_burst.py [hunk 9]
```diff
@@ -168,3 +168,28 @@ def test_get_burst_params_multiple_results():
...
+        burst.validate_bursts(
+            'S1_215032_IW2_20230802T144608_VV_7EE2-BURST',
+            'S1_215032_IW2_20230721T144607_HH_B3FA-BURST'
+        )
+    with pytest.raises(ValueError, match=r'.*do not share a common BurstID.*'):  # different burst ids
```
Comment (jtherrmann) [inline tests/test_burst.py:186]: The code is clear without the comment.
```suggestion
    with pytest.raises(ValueError, match=r'.*do not share a common BurstID.*'):
```

File: tests/test_burst.py [hunk 10]
```diff
@@ -168,3 +168,28 @@ def test_get_burst_params_multiple_results():
...
+        burst.validate_bursts(
+            'S1_030349_IW1_20230808T171601_VV_4A37-BURST',
+            'S1_030348_IW1_20230820T171602_VV_5AC3-BURST'
+        )
+    with pytest.raises(ValueError, match=r'.*only VV and HH.*'):  # incorrect polarizations
```
Comment (jtherrmann) [inline tests/test_burst.py:191]: The code is clear without the comment.
```suggestion
    with pytest.raises(ValueError, match=r'.*only VV and HH.*'):
```

File: images/coverage.svg
Comment (AndrewPlayer3) [reply images/coverage.svg:1]: It updated from 61% to 62%.

File: CHANGELOG.md [hunk 11]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.7.1]
+### Added
+* insar_tops_bursts now checks if burst pairs are valid, and that they have COP30 coverage.
```
Comment (jtherrmann) [inline CHANGELOG.md:11]: Where does the check for COP30 coverage take place?

Comment (jtherrmann) [inline CHANGELOG.md:11]:
```suggestion
* insar_tops_bursts now validates burst pair granule names.
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.7.1]
+### Added
+* insar_tops_bursts now validates burst pair granule names.
```
Comment (jtherrmann) [inline CHANGELOG.md:11]:
```suggestion
* insar_tops_burst now validates burst pair granule names.
```

**Review by jtherrmann — APPROVED** [review]

**Pull request merged**